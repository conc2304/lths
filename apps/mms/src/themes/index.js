import PropTypes from 'prop-types';
import { useMemo } from 'react';

import { CssBaseline } from '@mui/material';
import { createTheme, ThemeProvider } from '@mui/material/styles';

import getPalette from './palette';
import getTypography from './typography';
import getOverrides from './overrides';


export default function DashThemeProvider({ children }) {
    const theme = getPalette();

    const themeTypography = getTypography(`'Public Sans', sans-serif`);

    const themeOptions = useMemo(
        () => ({
            breakpoints: {
                values: {
                    xs: 0,
                    sm: 768,
                    md: 1024,
                    lg: 1266,
                    xl: 1536
                }
            },
            direction: 'ltr',
            mixins: {
                // layout header settings
                toolbar: {
                    minHeight: 50,
                    paddingTop: 4,
                    paddingBottom: 4
                },
            },
            palette: theme.palette,
            typography: themeTypography
        }),
        [theme, themeTypography]
    );

    const themes = createTheme(themeOptions);
    themes.components= getOverrides(themes);

    console.log('themes.components')
    console.log(themes.components)

    return (<ThemeProvider theme={themes}>
                <CssBaseline />
                {children}
            </ThemeProvider>);
}

DashThemeProvider.propTypes = {
    children: PropTypes.node
};
