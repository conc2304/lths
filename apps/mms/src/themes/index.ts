export { DashThemeProvider } from '@lths-mui/shared/mui-themes';
