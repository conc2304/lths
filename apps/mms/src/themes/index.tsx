import { ReactNode, useMemo } from 'react';

import { CssBaseline, Theme } from '@mui/material';
import { createTheme, ThemeProvider } from '@mui/material/styles';

import getPalette from './palette';
import getTypography from './typography';
import getComponentOverrides from './overrides/index';

type Props = {
    children : ReactNode;
}

export default function DashThemeProvider({ children }: Props) {
    const theme = getPalette();

    const themeTypography = getTypography(`'Public Sans', sans-serif`);

    const themeOptions: Theme = useMemo(
        () => ({
            breakpoints: {
                values: {
                    xs: 0,
                    sm: 768,
                    md: 1024,
                    lg: 1266,
                    xl: 1536
                }
            },
            direction: 'ltr',
            mixins: {
                // layout header settings
                toolbar: {
                    minHeight: 50,
                    paddingTop: 4,
                    paddingBottom: 4
                },
            },
            palette: theme.palette,
            ...themeTypography
        }),
        [theme, themeTypography]
    );

    const themes = createTheme(themeOptions);
    themes.components= getComponentOverrides(themes);

    return (<ThemeProvider theme={themes}>
                <CssBaseline />
                {children}
            </ThemeProvider>);
}


