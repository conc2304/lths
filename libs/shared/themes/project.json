{
  "name": "shared-themes",
  "$schema": "../../../node_modules/nx/schemas/project-schema.json",
  "sourceRoot": "libs/shared/themes/src",
  "projectType": "library",
  "tags": ["shared", "themes"],
  "targets": {
    "lint": {
      "executor": "@nx/linter:eslint",
      "outputs": ["{options.outputFile}"],
      "options": {
        "lintFilePatterns": ["libs/shared/themes/**/*.{ts,tsx,js,jsx}"]
      }
    }
  }
}
