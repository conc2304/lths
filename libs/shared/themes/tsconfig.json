{
  "compilerOptions": {
    "jsx": "react-jsx",
    "allowJs": false,
    "esModuleInterop": false,
    "allowSyntheticDefaultImports": true,
    "strict": true
  },
  "files": [
    "src/createPalette.d.ts",
    "src/Button.d.ts"
  ],
  "include": [],
  "references": [
    {
      "path": "./tsconfig.lib.json"
    }
  ],
  "extends": "../../../tsconfig.base.json"
}