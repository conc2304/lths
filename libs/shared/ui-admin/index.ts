export * from './src/feature-flags';
