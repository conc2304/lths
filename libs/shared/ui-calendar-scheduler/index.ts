export * from './src/lib/components/calendar/index';
export * from './src/lib/components/mock-events';
