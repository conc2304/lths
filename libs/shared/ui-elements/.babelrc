{
  "presets": [
    "@babel/preset-env",
    "@babel/preset-typescript",
    [
      "@nx/react/babel",
      {
        "runtime": "automatic",
        "useBuiltIns": "usage"
      }
    ]
  ],
  "plugins": []
}
