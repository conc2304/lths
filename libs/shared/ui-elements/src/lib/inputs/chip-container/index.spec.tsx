import React from 'react';
import { RBThemeProvider } from '@lths-mui/shared/themes';
import '@testing-library/jest-dom/extend-expect';
import { render, fireEvent, screen, within } from '@testing-library/react';

import { ChipContainer } from './index';

describe('ChipContainer component', () => {
  const onDeleteMock = jest.fn();
  const onClearAllMock = jest.fn();
  const openModalMock = jest.fn();

  const selectedFilters = {
    group1: {
      item1: { id: 'item1', title: 'Item 1' },
      item2: { id: 'item2', title: 'Item 2' },
    },
    group2: {
      item3: { id: 'item3', title: 'Item 3' },
    },
  };

  const itemsSelectedCount = Object.values(selectedFilters).reduce(
    (count, currentItem) => count + Object.entries(currentItem).length,
    0
  );

  beforeEach(() => {
    // Setup
  });

  afterEach(() => {
    // Clean up
    jest.clearAllMocks();
  });

  it('should match the snapshot (inline variant)', () => {
    // Arrange
    const component = (
      <ChipContainer
        title="Filters Applied"
        selectedFilters={selectedFilters}
        onDelete={onDeleteMock}
        variant="inline"
        onClearAll={onClearAllMock}
        openModal={openModalMock}
      />
    );

    const { baseElement } = render(RBThemeProvider({ children: component }));
    // Act
    // Assert
    expect(baseElement).toMatchSnapshot();
  });

  it('should match the snapshot (modal variant)', () => {
    // Arrange
    const component = (
      <ChipContainer
        title="Filters"
        onDelete={onDeleteMock}
        selectedFilters={selectedFilters}
        variant="modal"
        onClearAll={onClearAllMock}
        openModal={openModalMock}
      />
    );
    render(RBThemeProvider({ children: component }));

    const { baseElement } = render(RBThemeProvider({ children: component }));
    // Act
    // Assert
    expect(baseElement).toMatchSnapshot();
  });

  it('should render the ChipContainer with modal variant correctly', () => {
    // Arrange
    const component = (
      <ChipContainer
        title="Filters"
        onDelete={onDeleteMock}
        selectedFilters={selectedFilters}
        variant="modal"
        onClearAll={onClearAllMock}
        openModal={openModalMock}
      />
    );
    render(RBThemeProvider({ children: component }));

    // Act

    // Assert
    expect(screen.getByText('Filters')).toBeInTheDocument();
    expect(screen.getByText('Item 1')).toBeInTheDocument();
    expect(screen.getByText('Item 2')).toBeInTheDocument();
    expect(screen.getByText('Item 3')).toBeInTheDocument();
  });

  it('should render the ChipContainer with inline variant correctly', () => {
    // Arrange
    const component = (
      <ChipContainer
        title="Filters Applied"
        selectedFilters={selectedFilters}
        onDelete={onDeleteMock}
        variant="inline"
        onClearAll={onClearAllMock}
        openModal={openModalMock}
      />
    );
    render(RBThemeProvider({ children: component }));

    // Act

    // Assert
    expect(screen.getByText(`${itemsSelectedCount} Filters Applied`)).toBeInTheDocument();
    expect(screen.getByText('Clear All')).toBeInTheDocument();
  });

  it('should handle onDelete event correctly', () => {
    // Arrange
    const component = (
      <ChipContainer
        title="Filters"
        onDelete={onDeleteMock}
        selectedFilters={selectedFilters}
        variant="modal"
        onClearAll={onClearAllMock}
        openModal={openModalMock}
      />
    );
    render(RBThemeProvider({ children: component }));

    const testParentID = 'group1';
    const testItemID = 'item1';
    const testChip = selectedFilters[testParentID];
    const testChipLabel = testChip[testItemID].title;
    const testChipElem = screen.getByText(testChipLabel).parentElement as HTMLElement;

    const deleteButton = within(testChipElem).getByTestId('CloseIcon');

    // Act
    fireEvent.click(deleteButton);

    // Assert
    expect(deleteButton).toBeInTheDocument();
    expect(onDeleteMock).toHaveBeenCalledTimes(1);
    expect(onDeleteMock).toHaveBeenCalledWith({ parentID: testParentID, itemID: testItemID });
  });

  it('should handle onClearAll event correctly', () => {
    // Arrange
    const component = (
      <ChipContainer
        title="Filters"
        onDelete={onDeleteMock}
        selectedFilters={selectedFilters}
        variant="inline"
        onClearAll={onClearAllMock}
        openModal={openModalMock}
      />
    );
    render(RBThemeProvider({ children: component }));

    // Act
    fireEvent.click(screen.getByText('Clear All'));

    expect(onClearAllMock).toHaveBeenCalledTimes(1);
  });
});
