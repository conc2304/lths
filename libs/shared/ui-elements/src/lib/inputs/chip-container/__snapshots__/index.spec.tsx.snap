// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ChipContainer component should match the snapshot (inline variant) 1`] = `
<body>
  <div>
    <div>
      <div
        class="ChipContainer--content MuiBox-root css-egykz0"
      >
        <h5
          class="MuiTypography-root MuiTypography-h5 css-1hnqzd2-MuiTypography-root"
        >
          3
           
          Filters Applied
        </h5>
        <div
          class="ChipContainer--chips-wrapper MuiBox-root css-1k6yql2"
        >
          <div
            class="MuiButtonBase-root MuiChip-root MuiChip-filled MuiChip-sizeMedium MuiChip-colorDefault MuiChip-deletable MuiChip-deletableColorDefault MuiChip-filledDefault css-13bh5z-MuiButtonBase-root-MuiChip-root"
            role="button"
            tabindex="0"
          >
            <span
              class="MuiChip-label MuiChip-labelMedium css-1dybbl5-MuiChip-label"
            >
              Item 1
            </span>
            <svg
              aria-hidden="true"
              class="MuiSvgIcon-root MuiSvgIcon-fontSizeMedium MuiChip-deleteIcon MuiChip-deleteIconMedium MuiChip-deleteIconColorDefault MuiChip-deleteIconFilledColorDefault css-i4bv87-MuiSvgIcon-root"
              data-testid="CloseIcon"
              focusable="false"
              viewBox="0 0 24 24"
            >
              <path
                d="M19 6.41 17.59 5 12 10.59 6.41 5 5 6.41 10.59 12 5 17.59 6.41 19 12 13.41 17.59 19 19 17.59 13.41 12z"
              />
            </svg>
          </div>
          <div
            class="MuiButtonBase-root MuiChip-root MuiChip-filled MuiChip-sizeMedium MuiChip-colorDefault MuiChip-deletable MuiChip-deletableColorDefault MuiChip-filledDefault css-13bh5z-MuiButtonBase-root-MuiChip-root"
            role="button"
            tabindex="0"
          >
            <span
              class="MuiChip-label MuiChip-labelMedium css-1dybbl5-MuiChip-label"
            >
              Item 2
            </span>
            <svg
              aria-hidden="true"
              class="MuiSvgIcon-root MuiSvgIcon-fontSizeMedium MuiChip-deleteIcon MuiChip-deleteIconMedium MuiChip-deleteIconColorDefault MuiChip-deleteIconFilledColorDefault css-i4bv87-MuiSvgIcon-root"
              data-testid="CloseIcon"
              focusable="false"
              viewBox="0 0 24 24"
            >
              <path
                d="M19 6.41 17.59 5 12 10.59 6.41 5 5 6.41 10.59 12 5 17.59 6.41 19 12 13.41 17.59 19 19 17.59 13.41 12z"
              />
            </svg>
          </div>
          <div
            class="MuiButtonBase-root MuiChip-root MuiChip-filled MuiChip-sizeMedium MuiChip-colorDefault MuiChip-deletable MuiChip-deletableColorDefault MuiChip-filledDefault css-13bh5z-MuiButtonBase-root-MuiChip-root"
            role="button"
            tabindex="0"
          >
            <span
              class="MuiChip-label MuiChip-labelMedium css-1dybbl5-MuiChip-label"
            >
              Item 3
            </span>
            <svg
              aria-hidden="true"
              class="MuiSvgIcon-root MuiSvgIcon-fontSizeMedium MuiChip-deleteIcon MuiChip-deleteIconMedium MuiChip-deleteIconColorDefault MuiChip-deleteIconFilledColorDefault css-i4bv87-MuiSvgIcon-root"
              data-testid="CloseIcon"
              focusable="false"
              viewBox="0 0 24 24"
            >
              <path
                d="M19 6.41 17.59 5 12 10.59 6.41 5 5 6.41 10.59 12 5 17.59 6.41 19 12 13.41 17.59 19 19 17.59 13.41 12z"
              />
            </svg>
          </div>
        </div>
        <div>
          <button
            class="MuiButtonBase-root MuiButton-root MuiButton-text MuiButton-textSecondaryButton MuiButton-sizeLarge MuiButton-textSizeLarge MuiButton-root MuiButton-text MuiButton-textSecondaryButton MuiButton-sizeLarge MuiButton-textSizeLarge css-yxjgq5-MuiButtonBase-root-MuiButton-root"
            tabindex="0"
            type="button"
          >
            Clear All
            <span
              class="MuiTouchRipple-root css-8je8zh-MuiTouchRipple-root"
            />
          </button>
        </div>
      </div>
    </div>
  </div>
</body>
`;

exports[`ChipContainer component should match the snapshot (modal variant) 1`] = `
<body>
  <div>
    <div
      data-testid="chip-container-modal"
    >
      <div
        class="MuiBox-root css-1dufe52"
      >
        <h5
          class="MuiTypography-root MuiTypography-h5 css-jg55ep-MuiTypography-root"
        >
          Filters
        </h5>
        <span
          class="MuiBadge-root css-mel7su-MuiBadge-root"
        >
          <span
            class="MuiBadge-badge MuiBadge-standard MuiBadge-anchorOriginTopRight MuiBadge-anchorOriginTopRightRectangular MuiBadge-overlapRectangular MuiBadge-colorPrimary css-13dsch8-MuiBadge-badge"
          >
            3
          </span>
        </span>
      </div>
      <div
        class="MuiBox-root css-1o99rnv"
      >
        <div
          class="MuiButtonBase-root MuiChip-root MuiChip-filled MuiChip-sizeMedium MuiChip-colorDefault MuiChip-deletable MuiChip-deletableColorDefault MuiChip-filledDefault css-13bh5z-MuiButtonBase-root-MuiChip-root"
          role="button"
          tabindex="0"
        >
          <span
            class="MuiChip-label MuiChip-labelMedium css-1dybbl5-MuiChip-label"
          >
            Item 1
          </span>
          <svg
            aria-hidden="true"
            class="MuiSvgIcon-root MuiSvgIcon-fontSizeMedium MuiChip-deleteIcon MuiChip-deleteIconMedium MuiChip-deleteIconColorDefault MuiChip-deleteIconFilledColorDefault css-i4bv87-MuiSvgIcon-root"
            data-testid="CloseIcon"
            focusable="false"
            viewBox="0 0 24 24"
          >
            <path
              d="M19 6.41 17.59 5 12 10.59 6.41 5 5 6.41 10.59 12 5 17.59 6.41 19 12 13.41 17.59 19 19 17.59 13.41 12z"
            />
          </svg>
        </div>
        <div
          class="MuiButtonBase-root MuiChip-root MuiChip-filled MuiChip-sizeMedium MuiChip-colorDefault MuiChip-deletable MuiChip-deletableColorDefault MuiChip-filledDefault css-13bh5z-MuiButtonBase-root-MuiChip-root"
          role="button"
          tabindex="0"
        >
          <span
            class="MuiChip-label MuiChip-labelMedium css-1dybbl5-MuiChip-label"
          >
            Item 2
          </span>
          <svg
            aria-hidden="true"
            class="MuiSvgIcon-root MuiSvgIcon-fontSizeMedium MuiChip-deleteIcon MuiChip-deleteIconMedium MuiChip-deleteIconColorDefault MuiChip-deleteIconFilledColorDefault css-i4bv87-MuiSvgIcon-root"
            data-testid="CloseIcon"
            focusable="false"
            viewBox="0 0 24 24"
          >
            <path
              d="M19 6.41 17.59 5 12 10.59 6.41 5 5 6.41 10.59 12 5 17.59 6.41 19 12 13.41 17.59 19 19 17.59 13.41 12z"
            />
          </svg>
        </div>
        <div
          class="MuiButtonBase-root MuiChip-root MuiChip-filled MuiChip-sizeMedium MuiChip-colorDefault MuiChip-deletable MuiChip-deletableColorDefault MuiChip-filledDefault css-13bh5z-MuiButtonBase-root-MuiChip-root"
          role="button"
          tabindex="0"
        >
          <span
            class="MuiChip-label MuiChip-labelMedium css-1dybbl5-MuiChip-label"
          >
            Item 3
          </span>
          <svg
            aria-hidden="true"
            class="MuiSvgIcon-root MuiSvgIcon-fontSizeMedium MuiChip-deleteIcon MuiChip-deleteIconMedium MuiChip-deleteIconColorDefault MuiChip-deleteIconFilledColorDefault css-i4bv87-MuiSvgIcon-root"
            data-testid="CloseIcon"
            focusable="false"
            viewBox="0 0 24 24"
          >
            <path
              d="M19 6.41 17.59 5 12 10.59 6.41 5 5 6.41 10.59 12 5 17.59 6.41 19 12 13.41 17.59 19 19 17.59 13.41 12z"
            />
          </svg>
        </div>
      </div>
    </div>
  </div>
  <div>
    <div
      data-testid="chip-container-modal"
    >
      <div
        class="MuiBox-root css-1dufe52"
      >
        <h5
          class="MuiTypography-root MuiTypography-h5 css-jg55ep-MuiTypography-root"
        >
          Filters
        </h5>
        <span
          class="MuiBadge-root css-mel7su-MuiBadge-root"
        >
          <span
            class="MuiBadge-badge MuiBadge-standard MuiBadge-anchorOriginTopRight MuiBadge-anchorOriginTopRightRectangular MuiBadge-overlapRectangular MuiBadge-colorPrimary css-13dsch8-MuiBadge-badge"
          >
            3
          </span>
        </span>
      </div>
      <div
        class="MuiBox-root css-1o99rnv"
      >
        <div
          class="MuiButtonBase-root MuiChip-root MuiChip-filled MuiChip-sizeMedium MuiChip-colorDefault MuiChip-deletable MuiChip-deletableColorDefault MuiChip-filledDefault css-13bh5z-MuiButtonBase-root-MuiChip-root"
          role="button"
          tabindex="0"
        >
          <span
            class="MuiChip-label MuiChip-labelMedium css-1dybbl5-MuiChip-label"
          >
            Item 1
          </span>
          <svg
            aria-hidden="true"
            class="MuiSvgIcon-root MuiSvgIcon-fontSizeMedium MuiChip-deleteIcon MuiChip-deleteIconMedium MuiChip-deleteIconColorDefault MuiChip-deleteIconFilledColorDefault css-i4bv87-MuiSvgIcon-root"
            data-testid="CloseIcon"
            focusable="false"
            viewBox="0 0 24 24"
          >
            <path
              d="M19 6.41 17.59 5 12 10.59 6.41 5 5 6.41 10.59 12 5 17.59 6.41 19 12 13.41 17.59 19 19 17.59 13.41 12z"
            />
          </svg>
        </div>
        <div
          class="MuiButtonBase-root MuiChip-root MuiChip-filled MuiChip-sizeMedium MuiChip-colorDefault MuiChip-deletable MuiChip-deletableColorDefault MuiChip-filledDefault css-13bh5z-MuiButtonBase-root-MuiChip-root"
          role="button"
          tabindex="0"
        >
          <span
            class="MuiChip-label MuiChip-labelMedium css-1dybbl5-MuiChip-label"
          >
            Item 2
          </span>
          <svg
            aria-hidden="true"
            class="MuiSvgIcon-root MuiSvgIcon-fontSizeMedium MuiChip-deleteIcon MuiChip-deleteIconMedium MuiChip-deleteIconColorDefault MuiChip-deleteIconFilledColorDefault css-i4bv87-MuiSvgIcon-root"
            data-testid="CloseIcon"
            focusable="false"
            viewBox="0 0 24 24"
          >
            <path
              d="M19 6.41 17.59 5 12 10.59 6.41 5 5 6.41 10.59 12 5 17.59 6.41 19 12 13.41 17.59 19 19 17.59 13.41 12z"
            />
          </svg>
        </div>
        <div
          class="MuiButtonBase-root MuiChip-root MuiChip-filled MuiChip-sizeMedium MuiChip-colorDefault MuiChip-deletable MuiChip-deletableColorDefault MuiChip-filledDefault css-13bh5z-MuiButtonBase-root-MuiChip-root"
          role="button"
          tabindex="0"
        >
          <span
            class="MuiChip-label MuiChip-labelMedium css-1dybbl5-MuiChip-label"
          >
            Item 3
          </span>
          <svg
            aria-hidden="true"
            class="MuiSvgIcon-root MuiSvgIcon-fontSizeMedium MuiChip-deleteIcon MuiChip-deleteIconMedium MuiChip-deleteIconColorDefault MuiChip-deleteIconFilledColorDefault css-i4bv87-MuiSvgIcon-root"
            data-testid="CloseIcon"
            focusable="false"
            viewBox="0 0 24 24"
          >
            <path
              d="M19 6.41 17.59 5 12 10.59 6.41 5 5 6.41 10.59 12 5 17.59 6.41 19 12 13.41 17.59 19 19 17.59 13.41 12z"
            />
          </svg>
        </div>
      </div>
    </div>
  </div>
</body>
`;
